src/storage/store/engine.rs

```diff
@@ -947,15 +950,55 @@ impl ShardEngine {
                 // Validate ens
             }
             Some(proto::message_data::Body::VerificationAddAddressBody(add)) => {
-                let result = validations::validate_add_address(add, message_data.fid, network);
+                let result = verification::validate_add_address(add, message_data.fid, network);
```

Comment:
We should push all the branches of this match into `validations::message::validate_message` above. So that validate_message can be accessed independently via a validations crate, and not tied into the engine code. 

In hubble, these set of validations were for things that depended on external data or interdependencies between messages (e.g. validating ENS names had a match verification add)

Reply:
There are validations dependent on existing state, e.g. `validate_username` that queries the db for additional info – do we _want_ this in the validations methods?